tls/s2n_config.c

```diff
@@ -451,6 +451,13 @@ int s2n_config_set_check_stapled_ocsp_response(struct s2n_config *config, uint8_
     return 0;
 }
 
+int s2n_config_disable_x509_validity_period_validation(struct s2n_config *config)
```

Comment:
Should this take a boolean?

Reply:
Yeah, a boolean does seem more flexible. I changed it. I kind of liked the "disable" name without a boolean because it's clear from the name what the default behavior is, but I tried to make this clear in the documentation.

Reply:
I tentatively reverted adding the bool since I feel like the name without the bool is better, and it matches a similar existing API: 
- `s2n_config_disable_x509_verification`
- `s2n_config_disable_x509_time_verification`

But let me know if you think the bool is still worth adding and I will switch it back.